plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/extractor/FetchMetadataTest.java

```diff
@@ -161,29 +171,102 @@ public class MetadataCollectorTest implements RewriteTest {
                         </project>
                         """;
 
+    @Test
+    void testWithPomOnly() throws Exception {
```

Comment:
Miss one test with source file (no quark) to ensure only source that create metadata are marked. All are marked for now, which is incorrect